### ISSUE TITLE
Download URL is wrong

### ISSUE THREAD

**Issue by CodeMonk3y84** [open-circle-ltd/chocolatey.microsoft-office-deployment#28]
Please update the ODT download link.

**Comment by olifre** [comment]
@sbaerlocher Maybe requestiong a verification exemption (as discussed e.g. here: https://github.com/chocolatey/choco/issues/2520 ) would be best? 
If we are lucky, this may trigger the Chocolatey guys to give the verification error a closer look.

**Comment by sbaerlocher** [comment]
@olifre Thank you for providing a more detailed analysis. I retested the issue using the Local Vagrant Test Environment available at https://github.com/chocolatey-community/chocolatey-test-environment from Chocolatey, and was able to confirm that it works in this environment. I will inquire on the Chocolatey Discord channel to see if they can make an exception or provide further details.

**Comment by deargle** [comment]
download url changed again: today, April 11, 2023, a new version of the ODT, Version 16.0.16227.20258, was released by Microsoft -- see https://learn.microsoft.com/en-us/officeupdates/odt-release-history

New download url is https://download.microsoft.com/download/2/7/A/27AF1BE6-DD20-4CB4-B154-EBAB8A7D4A7E/officedeploymenttool_16227-20258.exe

**Comment by olifre** [comment]
@deargle Chocomilk has automatically updated to the new URL, it does check for new installers once every 24 hours:
https://github.com/open-circle-ltd/chocolatey.microsoft-office-deployment/commit/e11bcff08e73e23275dc8e45dc40e471ee23ed65

Interestingly, this version was finally approved by the Chocolatey repo maintainers. @sbaerlocher , do you understand why it suddenly "just works" again?

**Comment by deargle** [comment]
Great! No, I'm not sure why this one got approved.

**Comment by ThomasPatzig** [comment]
link to ODT has changed again:

https://download.microsoft.com/download/2/7/A/27AF1BE6-DD20-4CB4-B154-EBAB8A7D4A7E/officedeploymenttool_16501-20196.exe

do you think it is also possible to add the ODT binary itself direct into the choco package instead of update the url all time
... and have all time a period where this choco install must be fail?

I know it will be more size (from 8KiB to 3.5 MiB) but as long as there are no permalinks to ODT it will be more stable

**Comment by olifre** [comment]
> link to ODT has changed again:

Chocomilk has already updated to that:
https://github.com/open-circle-ltd/chocolatey.microsoft-office-deployment/commit/2e2e6a3c88f2b27f6aa52bb833b5d44c3ed90347

It will of course take a bit of time until it shows up in the Chocolatey repos, providing that this time, the packages don't time out during tests. 

> do you think it is also possible to add the ODT binary itself direct into the choco package [...]

I am quite sure Microsoft does not allow to mirror / distribute the binaries.

**Comment by ThomasPatzig** [comment]
of course ... you are right:

https://www.microsoft.com/en-us/legal/terms-of-use

_...
**Software**
...
Without limiting the foregoing, copying or reproduction of the software to any other server or location for further reproduction or redistribution is expressly prohibited, unless such reproduction or redistribution is expressly permitted by the license agreement accompanying such software.
..._

no way to add a 100% stable solution here.

**Comment by olifre** [comment]
> It will of course take a bit of time until it shows up in the Chocolatey repos, providing that this time, the packages don't time out during tests.

It seems the dreaded timeout has caused the package tests to fail once more :cry: .

**Comment by ThomasPatzig** [comment]
got it
last successful run has:

`...`
`2023-04-12 13:51:53,583 3884 [DEBUG] - Elevating permissions and running ["C:\ProgramData\chocolatey\lib\microsoft-office-deployment\tools\..\bin\setup.exe" /configure C:\ProgramData\chocolatey\lib\microsoft-office-deployment\tools\..\bin\Install.xml ]. This may take a while, depending on the statements.`
`2023-04-12 14:22:59,595 3884 [DEBUG] - Command ["C:\ProgramData\chocolatey\lib\microsoft-office-deployment\tools\..\bin\setup.exe" /configure C:\ProgramData\chocolatey\lib\microsoft-office-deployment\tools\..\bin\Install.xml ] exited with '0'.`
`2023-04-12 14:23:00,878 3884 [DEBUG] - Finishing 'Start-ChocolateyProcessAsAdmin'`
`2023-04-12 14:23:00,891 3884 [INFO ] - microsoft-office-deployment has been installed.`
`...`

but now it is:
`...`
`2023-06-13 10:09:14,981 3000 [DEBUG] - Elevating permissions and running ["C:\ProgramData\chocolatey\lib\microsoft-office-deployment\tools\..\bin\setup.exe" /configure C:\ProgramData\chocolatey\lib\microsoft-office-deployment\tools\..\bin\Install.xml ]. This may take a while, depending on the statements.`
`2023-06-13 10:54:06,224 3000 [WARN ] - Chocolatey timed out waiting for the command to finish. The timeout`
` specified (or the default value) was '2700' seconds. Perhaps try a`
` higher '--execution-timeout'? See 'choco -h' for details.'`
`...`

**Comment by ThomasPatzig** [comment]
I have install that version now many time:
https://community.chocolatey.org/packages/microsoft-office-deployment/16.0.16501.20196

and it was never blocked on my side like in that choco auto review process:
https://gist.github.com/choco-bot/7e55afaf84f7e6026f23f39e2bf864e8

Can you trigger the "Verification Testing" of "https://community.chocolatey.org/packages/microsoft-office-deployment/16.0.16501.20196" once again?

**Comment by olifre** [comment]
@sbaerlocher Maybe you can take a look at triggering verification testing once more? 

Or maybe @pauby can take a look at adding some exception? While Chocomilk seems to reliably bump the version of this package, verification tests appear to run into timeouts (likely the testing environment is "too slow" for MS Office installation testing in the time limit).

**Comment by pauby** [comment]
I have re-run the verification. If you need this re-run then adding a moderation comment on the package page is the best way to request it.

**Comment by olifre** [comment]
@pauby Many thanks! Since @ThomasPatzig and me are just "users" and not maintainers of the package, I am not sure we can add moderation comments (and the maintainers don't seem to be that responsive at the time being). 
Let's hope the re-run fixes things :smile: .

**Comment by pauby** [comment]
[That version](https://community.chocolatey.org/packages/microsoft-office-deployment/16.0.16501.20196) has now passed verification and has been approved!

**Comment by olifre** [comment]
@pauby Many thanks!

@CodeMonk3y84 If things also work for you, please close this issue. Chocomilk should update automatically as usually, if verification tests time out again the next time, we can open a new issue.

**Comment by DmitriiKh** [comment]
Thank you for your work.
Could you please update the ODT link again? 
https://download.microsoft.com/download/2/7/A/27AF1BE6-DD20-4CB4-B154-EBAB8A7D4A7E/officedeploymenttool_16626-20148.exe

**Comment by olifre** [comment]
@DmitriiKh Please don't 0-day-bump: @sbaerlocher has set up Chocomilk to run automatically once every day:
https://github.com/open-circle-ltd/chocolatey.microsoft-office-deployment/blob/6fe9415b3493223e209df7a8ecccf00899a23e7f/.github/workflows/chocomilk.yml#L6
It did run 8 hours before your comment and did not find the new version available yet. 

So we should check back tomorrow, and then can check whether the new version is found correctly, uploaded and approved automatically.

**Comment by DmitriiKh** [comment]
@olifre Thank you for your answer. I have just figured out why your CI/CD didn't pick the new version of ODT that was deployed yesterday. I took some time. Sorry about that.

Small suggestion: could you please add this information to README please? Like ODT links are getting updated automatically during a couple of days.

**Comment by olifre** [comment]
@DmitriiKh Sadly, I am also just a user of this package (like you), so I don't have any permissions to change things (otherwise, I would also have just re-triggered the CI pipeline after your comment). 

@sbaerlocher is maintaining this project, of course you could open a PR he might merge?

**Comment by ThomasPatzig** [comment]
ah... now I got it 

the chocomilk.yml observes via https://release-monitoring.org/project/21215/ for new version of ODT

latest version update there was seen 2023-08-01 16:33 UTC -> 16.0.16626.20148

but last run of chocomilk.yml  was 2023-08-01 15:16 GMT ~13:16 UTC and this had seen 
ODT 16.0.16529.20182 from 2023-07-14 18:00

next run of chocomilk.yml  will be ~ 2023-08-01 15:16 GMT ~13:16 UTC ....

**Comment by olifre** [comment]
@ThomasPatzig Exactly ;-). Chocomilk did it's commit here:
https://github.com/open-circle-ltd/chocolatey.microsoft-office-deployment/commit/8977d368c915eaed458474a6f822604acc431e5e
and the new version has been pushed to Chocolatey and entered automatic verification:
https://community.chocolatey.org/packages/microsoft-office-deployment/16.0.16626.20148

Sadly, automatic verification failed once more, with the usual cause (timeout), see:
https://gist.github.com/choco-bot/7fe40eaa8a2a8e6bea60ce170e3bf331#file-install-txt-L718-L720

@sbaerlocher has retriggered verification testing quite often in the past, and also done local testing (which always worked and took a few minutes). Sadly, it seems the automatic verification system is significantly slower and quite often runs into these timeouts. 
Since @sbaerlocher seems less active in the past months, I'm not sure he'll trigger it in the near future. 

Maybe @pauby can help out again? The last time he triggered it [in this very issue](https://github.com/open-circle-ltd/chocolatey.microsoft-office-deployment/issues/28#issuecomment-1616056824), this helped and the package stayed below the enforced timeout.

**Comment by pauby** [comment]
@olifre I've submitted this back to the package verifier again.

It may be the case that the maintainers of the `microsoft-office-deployment` package, like the `adobereader` package, are no longer maintaining it. My suggestion would be to go through the [triage](https://docs.chocolatey.org/en-us/community-repository/users/package-triage-process) process to look for a new maintainer.
